### PR TITLE
Issue1838 durack1 update cryptography 0.4 to 1.2.2

### LINKED ISSUE #1838: Add cryptography as dependency for MyProxyClient
@doutriaux1 it's likely cryptography needs to be added as a dependency of MyProxyClient [here](https://github.com/UV-CDAT/uvcdat/blob/master/CMake/cdat_modules/myproxyclient_deps.cmake#L1)

This is a downstream problem in https://github.com/PCMDI/pcmdi_metrics/issues/279#issuecomment-184896732

### PR DESCRIPTION
**Pull request by durack1**
Fix #1838 - Update cryptography 0.4 to 1.2.2 and correct MyProxyClient dependencies

### CODE REVIEW

**Comment by durack1** [comment]
@doutriaux1 currently testing this branch in `pcmdi_metrics` - will report back here

**Comment by durack1** [comment]
@doutriaux1 the onion continues to peel, will need to add `cffi`:
```
  Download error on https://pypi.python.org/simple/cffi/: [SSL:
  CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:590) -- Some
  packages may not be found!
  Download error on https://pypi.python.org/simple/: [SSL:
  CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:590) -- Some
  packages may not be found!
  No local packages or download links found for cffi>=1.4.1
```

**Comment by durack1** [comment]
@doutriaux1 and so it keeps going..
```
   From /export/durack1/160217_pcmdi_metrics/PCMDI_METRICS/tmp/uvcdat/uvcdat_build/CRYPTOGRAPHY-prefix/src/CRYPTOGRAPHY-stamp/CRYPTOGRAPHY-install-err.log
  -------------------------------------------------
  no previously-included directories found matching 'docs/_build'
  warning: no previously-included files matching '*' found under directory
  'vectors'
  build/temp.linux-x86_64-2.7/_openssl.c:1805: warning: function declaration
  isn\u2019t a prototype
  Download error on https://pypi.python.org/simple/ipaddress/: [SSL:
  CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:590) -- Some
  packages may not be found!
  Couldn't find index page for 'ipaddress' (maybe misspelled?)
  Download error on https://pypi.python.org/simple/: [SSL:
  CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:590) -- Some
  packages may not be found!
  No local packages or download links found for ipaddress
  error: Could not find suitable distribution for
  Requirement.parse('ipaddress')
```

File: CMake/cdat_modules/ipaddress_deps.cmake [hunk 1]
```diff
@@ -0,0 +1 @@
+set(IPADDRESS_deps ${pip_pkg} ${python_pkg} ${setuptools_pkg})
```
Comment (durack1) [inline CMake/cdat_modules/ipaddress_deps.cmake]: @doutriaux1 I'm not certain about the `pip_pkg` dependency, the `setup.py` contains:
```python
try:
    from setuptools import setup
except ImportError:
    from distutils.core import setup
```

File: CMake/cdat_modules/ipaddress_external.cmake [hunk 2]
```diff
@@ -0,0 +1,6 @@
+
+# create an external project to install MyProxyClient,
+# and configure and build it
+set(nm IPADDRESS)
+
+include(pipinstaller)
```
Comment (durack1) [inline CMake/cdat_modules/ipaddress_external.cmake]: @doutriaux1 I'm also not certain of the contents of this file

**Comment by durack1** [comment]
@doutriaux1 peeeeel..
```
   From /export/durack1/160217_pcmdi_metrics/PCMDI_METRICS/tmp/uvcdat/uvcdat_build/CRYPTOGRAPHY-prefix/src/CRYPTOGRAPHY-stamp/CRYPTOGRAPHY-install-err.log
  -------------------------------------------------
  no previously-included directories found matching 'docs/_build'
  warning: no previously-included files matching '*' found under directory
  'vectors'
  build/temp.linux-x86_64-2.7/_openssl.c:1805: warning: function declaration
  isn\u2019t a prototype
  Download error on https://pypi.python.org/simple/enum34/: [SSL:
  CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:590) -- Some
  packages may not be found!
  Couldn't find index page for 'enum34' (maybe misspelled?)
  Download error on https://pypi.python.org/simple/: [SSL:
  CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:590) -- Some
  packages may not be found!
  No local packages or download links found for enum34
  error: Could not find suitable distribution for Requirement.parse('enum34')
```

**Comment by durack1** [comment]
Peeeeeel..
```
   From /export/durack1/160217_pcmdi_metrics/PCMDI_METRICS/tmp/uvcdat/uvcdat_build/CRYPTOGRAPHY-prefix/src/CRYPTOGRAPHY-stamp/CRYPTOGRAPHY-install-err.log
  -------------------------------------------------
  no previously-included directories found matching 'docs/_build'
  warning: no previously-included files matching '*' found under directory
  'vectors'
  build/temp.linux-x86_64-2.7/_openssl.c:1805: warning: function declaration
  isn\u2019t a prototype
  Download error on https://pypi.python.org/simple/idna/: [SSL:
  CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:590) -- Some
  packages may not be found!
  Couldn't find index page for 'idna' (maybe misspelled?)
  Download error on https://pypi.python.org/simple/: [SSL:
  CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:590) -- Some
  packages may not be found!
  No local packages or download links found for idna>=2.0
  error: Could not find suitable distribution for
  Requirement.parse('idna>=2.0')
```

File: CMake/cdat_modules/enum34_external.cmake [hunk 3]
```diff
@@ -0,0 +1,6 @@
+
+# create an external project to install MyProxyClient,
+# and configure and build it
+set(nm ENUM34)
+
+include(pipinstaller)
```
Comment (durack1) [inline CMake/cdat_modules/enum34_external.cmake]: @doutriaux1 ? Not sure what this include does..

File: CMake/cdat_modules/idna_external.cmake [hunk 4]
```diff
@@ -0,0 +1,6 @@
+
+# create an external project to install MyProxyClient,
+# and configure and build it
+set(nm IDNA)
+
+include(pipinstaller)
```
Comment (durack1) [inline CMake/cdat_modules/idna_external.cmake]: @doutriaux1 ? Not sure what this include does..

**Comment by durack1** [comment]
@doutriaux1, ok this PR is good for review..

```
*******************************
UVCDAT  - issue1838_durack1_UpdateCrytography0.4To1.2.2 - Install Success
Metrics - climo_script - Install Success
*******************************
Please test as follows:
source /export/durack1/160217_pcmdi_metrics/PCMDI_METRICS/bin/setup_runtime.sh or .csh
python /export/durack1/160217_pcmdi_metrics/PCMDI_METRICS/tmp/metrics/test/test_suite.py
*******************************
Create your customized input_parameters.py (inspire yourself from examples in /export/durack1/160217_pcmdi_metrics/PCMDI_METRICS/doc/pcmdi_input_parameters_sample.py
Once you have a parameter file run:
source /export/durack1/160217_pcmdi_metrics/PCMDI_METRICS/bin/setup_runtime.sh or .csh
pcmdi_metrics_driver.py -p /path/to/your/edited/parameter_file.py
*******************************
Once everything is ok, you can safely remove the temporary directory: /export/durack1/160217_pcmdi_metrics/PCMDI_METRICS/tmp
*******************************
For further information or suggestions please contact the PCMDI Metrics Team @ pcmdi-metrics@llnl.gov
*******************************
```

**Comment by durack1** [comment]
@doutriaux1 looks like the linux bots have hung too..

**Comment by doutriaux1** [comment]
@durack1 thanks for this! Linux bots were not hang just so many commits lots in queue. Waiting for them and will review carefully and approve asap.

**Comment by doutriaux1** [comment]
oh you're right it says Thu not Fri... Ok will resumbit that and test on my mac

**Comment by durack1** [comment]
@doutriaux1 yes if you can take a look at the `*_deps.cmake` files and also the `*_external.cmake` files - I'm not 100% certain of how the info in these files are used.

**Comment by doutriaux1** [comment]
more peeling on my mac:
```

  ld: warning: directory not found for option
  '-L/lgm/uvcdat/2016-02-19_crypto/Externals/lib64'

  Download error on https://pypi.python.org/simple/pyasn1/: [SSL:
  CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:590) -- Some
  packages may not be found!

  Couldn't find index page for 'pyasn1' (maybe misspelled?)

  Download error on https://pypi.python.org/simple/: [SSL:
  CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:590) -- Some
  packages may not be found!

  No local packages or download links found for pyasn1>=0.1.8

  error: Could not find suitable distribution for
  Requirement.parse('pyasn1>=0.1.8')
```

**Comment by durack1** [comment]
@doutriaux1 I feel your pain.. But I hope I did get most of the dependencies sorted..

**Comment by durack1** [comment]
@doutriaux1 this is causing downstream issues for @gleckler1 in the pcmdi_metrics can it be bumped up in priority?

**Comment by doutriaux1** [comment]
@durack1 can you please add the missing dependencies? Thanks.

**Comment by durack1** [comment]
@doutriaux1 geez, assigning away all the jobs - you're not a deputy program manager by chance are you?.. I didn't hit any other problems in `linux`.. What OS did you hit problems on?

**Comment by durack1** [comment]
@doutriaux1 curiously the `travis` `OS X` builds completed fine ([here](https://travis-ci.org/UV-CDAT/uvcdat/jobs/110190917) and [here](https://travis-ci.org/UV-CDAT/uvcdat/jobs/110190914)).. So what's different about your env?

**Comment by doutriaux1** [comment]
mac 10.11 maybe? I always built fine on oceanonly to by the way, so it might be env dependent or machine load dependent

**Comment by durack1** [comment]
@doutriaux1 well your comment [above](https://github.com/UV-CDAT/uvcdat/pull/1840#issuecomment-187218051) doesn't really give me a specific target to work toward.. So am still awaiting specifics.. I have a `OS X 10.11.3` at home so can try that when I get back..

**Comment by durack1** [comment]
@doutriaux1 `pyasn1` in, so please do check this and merge ASAP.. It's causing grief in the `pcmdi_metrics` builds

**Comment by durack1** [comment]
@dnadeau4 can you review and merge please.
@aashish24 @doutriaux1 pinging you here too..

**Comment by durack1** [comment]
@doutriaux1 yep all good with `oceanonly` again - please review and merge ASAP:
```
*******************************
UVCDAT  - issue1838_durack1_UpdateCrytography0.4To1.2.2 - Install Success
Metrics - master - Install Success
*******************************
Please test as follows:
source /export/durack1/160302_pcmdi_metrics/PCMDI_METRICS/bin/setup_runtime.sh or .csh
python /export/durack1/160302_pcmdi_metrics/PCMDI_METRICS/tmp/metrics/test/test_suite.py
*******************************
Create your customized input_parameters.py (inspire yourself from examples in /export/durack1/160302_pcmdi_metrics/PCMDI_METRICS/doc/pcmdi_input_parameters_sample.py
Once you have a parameter file run:
source /export/durack1/160302_pcmdi_metrics/PCMDI_METRICS/bin/setup_runtime.sh or .csh
pcmdi_metrics_driver.py -p /path/to/your/edited/parameter_file.py
*******************************
Once everything is ok, you can safely remove the temporary directory: /export/durack1/160302_pcmdi_metrics/PCMDI_METRICS/tmp
*******************************
For further information or suggestions please contact the PCMDI Metrics Team @ pcmdi-metrics@llnl.gov
*******************************
```

**Comment by doutriaux1** [comment]
@durack1 giving it a shot on both my mac and ubuntu. Will let you know. 14 hours w/o a ping! What happened to you @durack1? Getting old :wink:?

**Comment by durack1** [comment]
@doutriaux1 ping!

**Comment by durack1** [comment]
@doutriaux1 can you double check the deps and Magic syntax pieces too.. @aashish24 has been silent

**Comment by doutriaux1** [comment]
@durack1 voila!

**Comment by durack1** [comment]
@doutriaux1 ok great, this fixes my issues for the `pcmdi_metrics` - you might want to ping Matthias to try again with the master branch..

**Comment by gleckler1** [comment]
@durack1 @doutriaux1 .... don't ping him yet - still bombing for me on oceanonly with cc

**Comment by durack1** [comment]
@doutriaux1 can you take a peek at @gleckler1's issue - looks like some weird `CMake` problem to me.. But what would I know..

**Pull request merged**